src/program.ts

```diff
@@ -2987,12 +3000,21 @@ export class File extends Element {
   }
 
   /** Creates an imported namespace from this file. */
-  asImportedNamespace(name: string, parent: Element, localIdentifier: IdentifierExpression): Namespace {
+  asImportedNamespace(
+    name: string, 
+    parent: Element, 
+    localIdentifier: IdentifierExpression
+  ): Namespace {
     var declaration = this.program.makeNativeNamespaceDeclaration(name);
     declaration.name = localIdentifier;
     var ns = new Namespace(name, parent, declaration);
     ns.set(CommonFlags.SCOPED);
     this.copyExportsToNamespace(ns);
+    // NOTE: Some exports are still queued, and are not added here.
+    // But will be added to the namespace when the exports are being resolved.
+    // We are adding this namespaces to our this files imported namespaces so
+    // we can resolve when we add our exports
+    this.importedNamespacesFromThisFile.push(ns);
```

Comment:
```suggestion
    // NOTE: Some exports are still queued, and can't yet be added here,
    // so we remember all the alias namespaces and add to them as well
    // when adding an element to the file.
    this.importedNamespacesFromThisFile.push(ns);
```

Indeed, this appears to be the most efficient way to handle it, thanks for updating! 👍